packages/grafana-ui/src/components/Forms/RadioButtonList/RadioButtonList.tsx

```diff
@@ -0,0 +1,56 @@
+import { css, cx } from '@emotion/css';
+import React from 'react';
+
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+
+import { useStyles2 } from '../../../themes';
+
+import { RadioButtonDot } from './RadioButtonDot';
+
+export interface RadioButtonListProps<T> {
+  name: string;
+  value?: T;
+  id?: string;
+  idSelector: (option: T) => string;
```

Comment:
why is this (`idSelector`) needed ?

Reply:
What if we force the user to pass a `SelectableValue & { id: string }`?

Reply:
E.g. `RadioButtonGroup` uses an array index to generate keys for options. As I know that's not the best solution. I wonder if we should use the `SelectableValue` interface at all here. Quite a lot of its fields is not used in this component

Reply:
Yea, and we do need a `value` (non-optional) – so perhaps a separate interface is better here? @peterholmberg what do you think?

Reply:
`SelectableValue` is not great but I'm not too fond of diverging from it in this PR.